src/ios/CordovaHttpPlugin.m

```diff
@@ -48,17 +96,30 @@ - (void)setupAuthChallengeBlock:(AFHTTPSessionManager*)manager {
         NSURLCredential * _Nullable __autoreleasing * _Nullable credential
     ) {
         if ([challenge.protectionSpace.authenticationMethod isEqualToString: NSURLAuthenticationMethodServerTrust]) {
-            *credential = [NSURLCredential credentialForTrust:challenge.protectionSpace.serverTrust];
 
-            if (![self->securityPolicy evaluateServerTrust:challenge.protectionSpace.serverTrust forDomain:challenge.protectionSpace.host]) {
-                return NSURLSessionAuthChallengeRejectProtectionSpace;
+            // Extract the server's trust
+            SecTrustRef serverTrust = challenge.protectionSpace.serverTrust;
+            NSString *serverDomain = challenge.protectionSpace.host;
+
+            // Retrieve the leaf certificate (index 0)
+            SecCertificateRef leafCertificate = SecTrustGetCertificateAtIndex(serverTrust, 0);
+
+            if (!leafCertificate) {
+                // If there is no leaf certificate, reject the connection
+                return NSURLSessionAuthChallengeCancelAuthenticationChallenge;
             }
 
-            if (credential) {
-                return NSURLSessionAuthChallengeUseCredential;
+            // Perform certificate pinning for root and intermediate certificates only if CN matches
+            if (![self->securityPolicy evaluateServerTrust:serverTrust forDomain:serverDomain]) {
```

Comment:
In evaluateServerTrust:
We should add the layer for the domains.

Configure per domain, if pinned or not.

Sample code for the idea
```
       // only cert pinning, if domain is configured for pinning
        if (isDomainPinned) {
          for (NSData *trustChainCertificate in [serverCertificates reverseObjectEnumerator]) {
            if ([self.pinnedCertificates containsObject:trustChainCertificate]) {
              return YES;
            }
          }
            
          return NO;
        }
        
        return YES;
```